stdlib/asm/crypto/dsa/rpo_falcon512.masm

```diff
@@ -218,182 +207,83 @@ export.load_h_s2_and_product.4
         push.M u32lt assert
         push.M u32lt assert
 
+        horner_eval_base
```

Comment:
can we also add some comments for the state of the stack in this loop? It's quite hard to follow